flygym/examples/plume_tracking/arena.py

```diff
@@ -0,0 +1,167 @@
+from multiprocessing import shared_memory
+
+from flygym.arena import BaseArena
+from flygym import NeuroMechFly
+
+
+class OdorPlumeArena(BaseArena):
+    def __init__(
+        self,
+        plume_data_path: Path,
+        dimension_scale_factor: float = 0.5,
+        plume_simulation_fps: float = 200,
+        intensity_scale_factor: float = 1.0,
+        friction: Tuple[float, float, float] = (1, 0.005, 0.0001),
+        num_sensors: int = 4,
+    ):
+        super().__init__()
+
+        self.dimension_scale_factor = dimension_scale_factor
+        self.plume_simulation_fps = plume_simulation_fps
+        self.intensity_scale_factor = intensity_scale_factor
+        self.friction = friction
+        self.num_sensors = num_sensors
+        self.curr_time = 0
+        self.plume_update_interval = 1 / plume_simulation_fps
+
+        # Load plume data
+        self.plume_datset = h5py.File(plume_data_path, "r")
+        self.plume_grid = self.plume_datset["plume"]
+        self.arena_size = (
+            np.array(self.plume_grid.shape[1:][::-1]) * dimension_scale_factor
+        )
+
+        # Set up floor
+        floor_material = self.root_element.asset.add(
+            "material",
+            name="floor_material",
+            reflectance=0.0,
+            shininess=0.0,
+            specular=0.0,
+            rgba=[0.6, 0.6, 0.6, 1],
+        )
+        self.root_element.worldbody.add(
+            "geom",
+            name="floor",
+            type="box",
+            size=(self.arena_size[0] / 2, self.arena_size[1], 1),
+            pos=(self.arena_size[0] / 2, self.arena_size[1] / 2, -1),
+            material=floor_material,
+        )
+
+        # Add birdeye camera
+        self.birdeye_cam = self.root_element.worldbody.add(
+            "camera",
+            name="birdeye_cam",
+            mode="fixed",
+            pos=(
+                0.50 * self.arena_size[0],
+                0.15 * self.arena_size[1],
+                1.00 * self.arena_size[1],
+            ),
+            euler=(np.deg2rad(15), 0, 0),
+            fovy=60,
+        )
+
+        # # Add second camera
+        # self.birdeye_cam = self.root_element.worldbody.add(
+        #     "camera",
+        #     name="birdeye_cam",
+        #     mode="fixed",
+        #     pos=(158, 93, 20),
+        #     euler=(0, 0, 0),
+        #     fovy=45,
+        # )
+
+    def get_position_mapping(
+        self, sim: NeuroMechFly, camera_id: str = "birdeye_cam"
+    ) -> np.ndarray:
+        """Get the display location (row-col coordinates) of each pixel on
+        the fluid dynamics simulation.
+
+        Parameters
+        ----------
+        sim : NeuroMechFly
+            NeuroMechFly simulation object.
+        camera_id : str, optional
+            Camera to build position mapping for, by default "birdeye_cam"
+
+        Returns
+        -------
+        pos_display: np.ndarray
+            Array of shape (n_row_pxls_plume, n_row_pxls_plume, 2)
```

Comment:
Not sure but should that not be n_cols_pxls_plume ?

Reply:
You are right! fixed directly in 26e2656